File: server2.js
const express = require("express");
const { io } = require("socket.io-client");
const axios = require("axios");
const cors = require("cors");
const sharp = require("sharp");
const fs = require("fs");

const app = express();
const iosocket = io("http://localhost:3333");

iosocket.on("connect", () => {
  iosocket.on("server1 server2", async (data) => {
    console.log("Processando imagem...");
    const colorImages = [];
    const { id, images } = data;

    for (let image of images) {
      try {
        const { data } = await axios.get(image); // Busca imagem enviando requisição para retorno do Buffer
        colorImages.push(data);
      } catch (error) {}
    }

    const result = [];
    for (let image of colorImages) {
      const newImage = await sharp(image).grayscale().toBuffer(); // Realiza o processamento em cada imagem
      result.push(newImage);
    }

    console.log("Processo finalizado. Enviando imagem de volta...");
    iosocket.emit(
      "server2 server1",
      { id, images: result.map((i) => i.toString("base64")) } // Envia imagens no formato base64
    );
  });
});

app.listen(3334);

File: server2.py
import socketio
import io
import base64
from PIL import Image

sio = socketio.Client()
sio.connect('http://localhost:3333')


def get_base64_code(images: list[str]) -> list[str]:
    result = []
    for image in images:
        string = ''
        for i in range(22, len(image)):
            string += image[i]
        result.append(string)
    return result


@sio.on('server1 server2')
def process_images(images):
    print('Processando imagem...')
    images = get_base64_code(images)

    result = []
    for image in images:
        img_bytes = base64.b64decode(image)
        img = Image.open(io.BytesIO(img_bytes))
        img.convert('L')

        with io.BytesIO() as buffer:
            img.save(buffer, format='PNG')
            result.append(buffer.getvalue())

    print('Processo finalizado. Enviando imagem de volta...')
    sio.emit('server2 server1', [base64.b64encode(img).decode('utf-8') for img in result])
